src/lib_json/json_reader.cpp

```diff
@@ -1268,6 +1271,13 @@ void OurReader::skipSpaces() {
   }
 }
 
+void OurReader::skipBom() {
+  if (strncmp(begin_, "\xEF\xBB\xBF", 3) == 0) {
```

Comment:
We need to check the input length first, in case it is shorter than 3 bytes.

I think `{}` is ok since there is always a trailing null or EOF. But `9` is a valid JSON document. Not sure what happens with empty-string, but that's probably checked elsewhere anyway.

Reply:
Yes, I considered its length, but later I realized that: 1) the function `strncmp()` allow its first parameter shorter than its second, this is safe. 2) if the input length is shorter than 3 bytes, the `strncmp() < 0`, it won't be bom characters so it won't do anything, this is safe too.

Reply:
Oh, sorry. You're right.